### PR TITLE
adding oldest_create_index

### PR DESCRIPTION
**Pull request by flipyap**
Adding an "oldest_create_index" node selector to the list of selectors to help with behaviors similar to that of the default AWS scaling strategy.

### CODE REVIEW

**Comment by tgross** [comment]
Hi @flipyap! We're going to take a look at this soon!

(Internal ref: https://hashicorp.atlassian.net/browse/NET-10944)

File: sdk/helper/scaleutils/nodeselector/oldest_create_index_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"testing"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_oldestClusterScaleInNodeSelectorName(t *testing.T) {
+	assert.Equal(t, "oldest_create_index", newOldestCreateIndexClusterScaleInNodeSelector().Name())
```
Comment (tgross) [inline sdk/helper/scaleutils/nodeselector/oldest_create_index_test.go:14]: We're trying to gradually migrate Nomad repos to use `shoenig/test` over testify:

```suggestion
	must.Eq(t, "oldest_create_index", newOldestCreateIndexClusterScaleInNodeSelector().Name())
```

File: sdk/helper/scaleutils/nodeselector/oldest_create_index_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+				{ID: "8d01d8fe-3c6b-0fdd-ebbc-a8b5cbb9c00c"},
+			},
+			name: "multiple selection",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actualOutput := newOldestCreateIndexClusterScaleInNodeSelector().Select(tc.inputNodes, tc.inputNum)
+			assert.Equal(t, tc.expectedOutput, actualOutput, tc.inputNodes)
```
Comment (tgross) [inline sdk/helper/scaleutils/nodeselector/oldest_create_index_test.go:53]:
```suggestion
			must.Eq(t, tc.expectedOutput, actualOutput)
```

File: sdk/helper/scaleutils/nodeselector/oldest_create_index_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nodeselector
+
+import (
+	"testing"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/stretchr/testify/assert"
```
Comment (tgross) [inline sdk/helper/scaleutils/nodeselector/oldest_create_index_test.go:10]:
```suggestion
	"github.com/shoenig/test/must"
```

**Review by tgross — CHANGES_REQUESTED** [review]
Hi @flipyap! I've left some minor comments but otherwise this looks great! Can you add a line to the [CHANGELOG.md](https://github.com/hashicorp/nomad-autoscaler/blob/main/CHANGELOG.md#unreleased) as well?

Once this ships, I'll follow-up with a docs PR [here]( website/content/tools/autoscaling/concepts/policy-eval/node-selector-strategy.mdx) in the main Nomad repo.

**Review by tgross — APPROVED** [review]
LGTM! Thanks @flipyap!

I've resolved merge conflicts with the changelog via the Github UI because you don't have edits enabled on your branch. Once CI is green I'll merge this.

**Comment by flipyap** [comment]
Thanks!

**Pull request merged**